### DISCUSSION TITLE
Multiple Windows 10 clients behind same NAT network

### DISCUSSION THREAD

**Discussion by DavorJ** [strongswan/strongswan#1035] [Q&A] [answered]
I have two Windows 10 clients behind a router. Any one of the two can connect to the VPN server. After one is connected, if I try to connect with the other one, it fails with RasClient EventID 20227: 

> The user SYSTEM dialed a connection named xxxxxxxx.com which has failed. The error code returned on failure is 809.

On the server side I see the following:

```
May 04 21:42:27 SERVER charon-systemd[16806]: received packet: from xxx.xxx.xxx.xxx[xx] to zzz.zzz.zzz.zzz[zzz] (632 bytes)
May 04 21:42:27 SERVER charon-systemd[16806]: parsed IKE_SA_INIT request 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V V V ]
May 04 21:42:27 SERVER charon-systemd[16806]: received MS NT5 ISAKMPOAKLEY v9 vendor ID
May 04 21:42:27 SERVER charon-systemd[16806]: received MS-Negotiation Discovery Capable vendor ID
May 04 21:42:27 SERVER charon-systemd[16806]: received Vid-Initial-Contact vendor ID
May 04 21:42:27 SERVER charon-systemd[16806]: received unknown vendor ID: xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx:xx
May 04 21:42:27 SERVER charon-systemd[16806]: xxx.xxx.xxx.xxx is initiating an IKE_SA
May 04 21:42:27 SERVER charon-systemd[16806]: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
May 04 21:42:27 SERVER charon-systemd[16806]: local host is behind NAT, sending keep alives
May 04 21:42:27 SERVER charon-systemd[16806]: remote host is behind NAT
May 04 21:42:27 SERVER charon-systemd[16806]: sending cert request for "C=XX, O=XXXXXX, CN=XX"
May 04 21:42:27 SERVER charon-systemd[16806]: sending cert request for "C=XX, O=XXXXXX, CN=XXXXXXXXXXXX"
May 04 21:42:27 SERVER charon-systemd[16806]: generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(MULT_AUTH) ]
May 04 21:42:27 SERVER charon-systemd[16806]: sending packet: from zzz.zzz.zzz.zzz[zzz] to xxx.xxx.xxx.xxx[xx] (493 bytes)
May 04 21:42:47 DEBIAN charon-systemd[16806]: sending keep alive to xxx.xxx.xxx.xxx[xx]
May 04 21:42:57 DEBIAN charon-systemd[16806]: deleting half open IKE_SA with xxx.xxx.xxx.xxx after timeout
```

Googling one ends up at two possible causes:
1) Fragmentation. This seems an old issue. Apparently since Windows (Server?) 1803 fragmentation is supported. Anyhow, based on the above logs, I don't see any fragmentation. And also setting `fragmentation = no` option makes no difference.
2) The fact that multiple clients are behind a NAT. There seems to be a solution for [L2TP/IPsec](https://docs.microsoft.com/en-us/troubleshoot/windows-server/networking/configure-l2tp-ipsec-server-behind-nat-t-device), but this [is not supposed to work](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients.md#windows-error-809) for IKEv2. Setting `encap = yes` has no effect.

**Is there a solution to this problem I am not finding?** 

_Note_: 1 Windows + 1 Android client can both connect at the same time. So it is mainly 2 Windows machines that are exclusive.

**Comment by tobiasbrunner** [comment, marked as answer]
The client either does not send the IKE_AUTH message or it doesn't get through to the server. The latter could be due to IP fragmentation. However, both peers support IKEv2 fragmentation (`FRAG_SUP` notify), so that might not be it (it could still happen if the IKEv2 fragments are too big). It also doesn't explain how a second client behind the same NAT should affect this (unless the NAT router is doing something weird). You could check on the client with Wireshark (or some other capturing tool) if it actually sends an IKE_AUTH message.

**Reply by DavorJ** [reply]
Seems you are correct. I just tested it with Android 11 (Kernel 4.14.190) using G4 and WiFi hotspot, and both Win 10 clients can connect without an issue through the hotspot. So no strongSwan issue :thumbsup:.
